### PR TITLE
First draft at adding cookbook to monofony docs

### PR DESCRIPTION
**Pull request by lbaey-mobizel**
This is the sortable recipe, coming next is translatable

### CODE REVIEW

File: docs/cookbook.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Chapter 9 - Cookbook
+
+Some recipes ou might enjoy to enhance Monofony easily according to your functional needs.
```
Comment (loic425) [inline docs/cookbook.md]:
```suggestion
Some recipes you might enjoy to enhance Monofony easily according to your functional needs.
```

File: docs/cookbook/sortable.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
+# 9.1 - Basics
+
+## Update your composer configuration
+
+We'll need doctrine extensions
```
Comment (loic425) [inline docs/cookbook/sortable.md]:
```suggestion
We'll need Doctrine extensions
```

File: docs/cookbook/sortable.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                'label' => 'app.position',
+            ])
+        ;
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'data_class' => Book::class,
+            'csrf_protection' => false,
```
Comment (loic425) [inline docs/cookbook/sortable.md:109]: hum, could we handle this?

**Review by loic425 — CHANGES_REQUESTED** [review]

File: docs/cookbook/sortable.md (hunk 3 above)
Comment (lbaey-mobizel) [reply docs/cookbook/sortable.md:109]: How so ?

File: docs/cookbook/sortable.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+```php
+# src/Grid/BookGrid.php
+$gridBuilder
+    ->orderBy('position', 'asc')
+    ->addField(
+        TwigField::create('position', 'backend/book/grid/field/positionWithButtons.html.twig')
+        ->setLabel('sylius.ui.position')
+        ->setPath('.')
+        ->setSortable(true),
```
Comment (loic425) [inline docs/cookbook/sortable.md]:
```suggestion
            ->setLabel('sylius.ui.position')
            ->setPath('.')
            ->setSortable(true),
```

File: docs/cookbook/sortable.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+# 9.3 - Bringing the sorting ability to backend grid
+
+## Add field to entity grid
+
+```php
+# src/Grid/BookGrid.php
+$gridBuilder
+    ->orderBy('position', 'asc')
+    ->addField(
+        TwigField::create('position', 'backend/book/grid/field/positionWithButtons.html.twig')
```
Comment (loic425) [inline docs/cookbook/sortable.md]:
```suggestion
        TwigField::create('position', 'backend/book/grid/field/position_with_buttons.html.twig')
```
Sylius does not follow the Symfony best practices. Symfony recommands to use snake case for templates names.

File: docs/cookbook/sortable.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+        ->setLabel('sylius.ui.position')
+        ->setPath('.')
+        ->setSortable(true),
+    )
+```
+
+## Add new grid field template
+
+```html
+<!-- template/backend/book/grid/field/positionWithButtons.html.twig -->
```
Comment (loic425) [inline docs/cookbook/sortable.md]:
```suggestion
<!-- template/backend/book/grid/field/position_with_buttons.html.twig -->
```

File: docs/cookbook/sortable.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class BookPositionType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('position', IntegerType::class, [
+                'label' => 'app.position',
```
Comment (loic425) [inline docs/cookbook/sortable.md]:
```suggestion
                'label' => 'sylius.ui.position',
```

File: docs/cookbook/sortable.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+    order: ['position' => 'ASC'],
+)]
+```
+
+
+# 9.4 Updating tests
+
+- Update your `BookFactory` class to add position in your fixtures.
+- Add functional tests according to the purpose of sorting in your app.
+- If using ApiPlatform, update your previous tests results with the new entity sorting.
```
Comment (loic425) [inline docs/cookbook/sortable.md]:
```suggestion
- If you use ApiPlatform, update your previous tests results with the new entity sorting.
```

File: docs/cookbook/translatable.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+Declare your future translation entity which will contain all the translations for every translatable field.
+```yaml
+sylius_resource:
+  resources:
+    app.book:
+      classes:
+        model: App\Entity\Book
+      translation:
+        classes:
+          model: App\Entity\BookTranslation
```
Comment (loic425) [inline docs/cookbook/translatable.md]:
```suggestion
sylius_resource:
    resources:
        app.book:
            classes:
                model: App\Entity\Book
            translation:
                classes:
                    model: App\Entity\BookTranslation
```

File: docs/cookbook/translatable.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+Say you want tu localize the book property `title`.
```
Comment (loic425) [inline docs/cookbook/translatable.md]:
```suggestion
Let's localize the book property `title`.
```

File: docs/cookbook/translatable.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    public function setTitle(string $title): void
+    {
+        $this->title = $title;
+    }
+}
+```
+
+And update your `Book` entity class so :
```
Comment (loic425) [inline docs/cookbook/translatable.md:53]:
```suggestion
And update your `Book` entity class:
```

**Review by loic425 — CHANGES_REQUESTED** [review]

**Pull request merged**